src/terminal/parser/stateMachine.cpp

```diff
@@ -25,6 +25,9 @@ StateMachine::StateMachine(std::unique_ptr<IStateMachineEngine> engine, const bo
     _oscString{},
     _cachedSequence{ std::nullopt }
 {
+    // The state machine must always accept C1 controls for the input engine,
+    // otherwise it won't work when the ConPTY terminal has S8C1T enabled.
+    _parserMode.set(Mode::AcceptC1, _isEngineForInput);

```

Comment:
Why don't we check whether S8C1T / `SendC1` is enabled? Since ConPTY parses all VT output, shouldn't it be aware over whether the mode is enabled or not?

Do we need to prevent a console application from disabling this mode?

Reply:
> Why don't we check whether S8C1T / `SendC1` is enabled? Since ConPTY parses all VT output, shouldn't it be aware over whether the mode is enabled or not?

Hypothetically, the conpty terminal could enable S8C1T as a user preference (that was certainly an option on the original hardware terminals). But even if that wasn't possible, it just seems like an unnecessary complication to make it dependent on the mode. If we received a C1 control sequence when we weren't expecting one, would we ever want to interpret that as something else?

> Do we need to prevent a console application from disabling this mode?

That shouldn't be possible. The only time we change parser modes at runtime is with the output state machine.

Reply:
This explanation tracks for me. Thanks!